src/coreclr/inc/executableallocator.h

```diff
@@ -254,7 +254,7 @@ class ExecutableWriterHolder
 
     void Unmap()
     {
-#if defined(HOST_OSX) && defined(HOST_ARM64) && !defined(DACCESS_COMPILE)
+#if defined(HOST_APPLE) && defined(HOST_ARM64) && !defined(DACCESS_COMPILE)
```

Comment:
I wonder about many of the changes to HOST_APPLE in coreclr stuff. This code is never compiled for anything else than macOS on Apple, so it seems these changes are unnecessary.

Reply:
The rationale was that it is not easy to tell at a glance which parts are used by nativeaot and which aren't (especially in these common include folders) so it was more straightforward to change them all since it doesn't hurt. I can revert them if you want.

Reply:
I have believed that nativeaot runtime code is contained in the nativeaot folder and doesn't include anything out of that. Isn't it the case anymore? 
Global changes like this are always causing me pain when backporting fixes to older releases, since you usually want the backport changes to be as small as possible. If you touch tens of files just to change an ifdef without a real need to fix something, then if I e.g. need to make a fix to just the executableallocator.cpp/.h (which I did recently), I would have to either backport all the extra files or manually pick a portion of them and risk breaking something. So I'd prefer reverting the change where it is not needed.

Reply:
> I have believed that nativeaot runtime code is contained in the nativeaot folder and doesn't include anything out of that. Isn't it the case anymore?

I think that's largely still the case but there are exceptions like the `gc`.

Reply:
ok, I'll revert the unnecessary changes then

Reply:
> I think that's largely still the case but there are exceptions like the `gc`.

Right, I meant the runtime without the GC.

Reply:
Yup. I just checked and here's the list of files outside of the `nativeaot` or `gc` directories that we depend on from the NativeAOT build (extracted from CMake):
- src/coreclr/inc/gcinfodecoder.h
- src/coreclr/inc/gcinfotypes.h
- src/coreclr/inc/gcmsg.inl
- src/coreclr/pal/inc/rt/no_sal2.h
- src/coreclr/pal/inc/rt/sal.h
- src/coreclr/pal/inc/rt/specstrings.h
- src/coreclr/pal/inc/rt/specstrings_strict.h
- src/coreclr/vm/common.h
- src/coreclr/vm/gcinfodecoder.cpp
- src/native/external/llvm-libunwind/include/__libunwind_config.h
- src/native/external/llvm-libunwind/include/libunwind.h
- src/native/external/llvm-libunwind/include/mach-o/compact_unwind_encoding.h
- src/native/external/llvm-libunwind/include/unwind.h
- src/native/external/llvm-libunwind/include/unwind_itanium.h
- src/native/external/llvm-libunwind/src/AddressSpace.hpp
- src/native/external/llvm-libunwind/src/CompactUnwinder.hpp
- src/native/external/llvm-libunwind/src/DwarfInstructions.hpp
- src/native/external/llvm-libunwind/src/DwarfParser.hpp
- src/native/external/llvm-libunwind/src/EHHeaderParser.hpp
- src/native/external/llvm-libunwind/src/RWMutex.hpp
- src/native/external/llvm-libunwind/src/Registers.hpp
- src/native/external/llvm-libunwind/src/Unwind-EHABI.cpp
- src/native/external/llvm-libunwind/src/Unwind-EHABI.h
- src/native/external/llvm-libunwind/src/UnwindCursor.hpp
- src/native/external/llvm-libunwind/src/Unwind_AppleExtras.cpp
- src/native/external/llvm-libunwind/src/cet_unwind.h
- src/native/external/llvm-libunwind/src/config.h
- src/native/external/llvm-libunwind/src/dwarf2.h
- src/native/external/llvm-libunwind/src/libunwind.cpp
- src/native/external/llvm-libunwind/src/libunwind_ext.h
- src/native/minipal/utils.h

Reply:
I reverted the unnecessary changes.

Reply:
Dependency on vm/common.h looks suspicious, it basically pulls in majority of the coreclr headers. Where do we include it in NativeAOT?

Reply:
@janvorli the chain is like this from what I can see:
- https://github.com/dotnet/runtime/blob/3cb2dae6f1520bf4b1e4658889f83a8cccc32ade/src/coreclr/nativeaot/Runtime/unix/UnixNativeCodeManager.cpp#L18
- https://github.com/dotnet/runtime/blob/3cb2dae6f1520bf4b1e4658889f83a8cccc32ade/src/coreclr/nativeaot/Runtime/gcinfodecoder.cpp#L6
- https://github.com/dotnet/runtime/blob/3cb2dae6f1520bf4b1e4658889f83a8cccc32ade/src/coreclr/vm/gcinfodecoder.cpp#L4

Reply:
FWIW I tried ifdefing out the common.h include and my iOS app still works.

Reply:
just including `gcinfodecoder.h` instead of the .cpp works too

Reply:
ah, I think this is just CMake's tracking having a false positive here. We define `_common_h_` in `nativeaot/Runtime/gcinfodecoder.cpp` so the header file won't actually be included in the `vm/gcinfodecoder.cpp`